File: backend/src/main/kotlin/top/bearingwall/asya/model/Instruction.kt
package top.bearingwall.asya.model

import jakarta.persistence.*
import top.bearingwall.asya.util.LocalDateTimeStringConverter
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "instructions")
class Instruction(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var uuid: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conference_id", nullable = false)
    var conference: Conference,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "submitter_id", nullable = false)
    var submitter: User,

    @Column(nullable = false, length = 200)
    var title: String,

    @Enumerated(EnumType.STRING)
    @Column(name = "instruction_type", nullable = false, length = 32)
    var instructionType: InstructionType,

    @Lob
    @Column(name = "instruction_content", nullable = false, columnDefinition = "TEXT")
    var content: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 32)
    var status: InstructionStatus = InstructionStatus.SUBMITTED,

    @Lob
    @Column(name = "review_comment", columnDefinition = "TEXT")
    var reviewComment: String? = null,

    @Column(name = "submit_real_time", nullable = false)
    var submitRealTime: LocalDateTime,

    @Column(name = "submit_game_time", nullable = false)
    @Convert(converter = LocalDateTimeStringConverter::class)
    var submitGameTime: LocalDateTime,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reviewed_by")
    var reviewedBy: User? = null,

    @Column(name = "reviewed_real_time")
    var reviewedRealTime: LocalDateTime? = null,

    @Column(name = "reviewed_game_time")
    @Convert(converter = LocalDateTimeStringConverter::class)
    var reviewedGameTime: LocalDateTime? = null
)

File: backend-dotnet/src/AsyaMun.Api/Models/Core/Instruction.cs
namespace AsyaMun.Api.Models;

public class Instruction
{
    public Guid Uuid { get; set; }

    public string InstructionContent { get; set; } = null!;

    public InstructionType InstructionType { get; set; }

    public string? ReviewComment { get; set; }

    public string? ReviewedGameTime { get; set; }

    public DateTime? ReviewedRealTime { get; set; }

    public string SubmitGameTime { get; set; } = null!;

    public DateTime SubmitRealTime { get; set; }

    public InstructionStatus Status { get; set; }

    public string Title { get; set; } = null!;

    public Guid ConferenceId { get; set; }

    public Guid? ReviewedBy { get; set; }

    public Guid SubmitterId { get; set; }

    public virtual Conference Conference { get; set; } = null!;

    public virtual User? ReviewedByNavigation { get; set; }

    public virtual User Submitter { get; set; } = null!;
}